src/librustc/ty/mod.rs

```diff
@@ -2282,6 +2282,13 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         }
     }
 
+    // Hygienically compare a use-site name (`use_name`) for a field or an associated item with its
+    // supposed definition name (`def_name`). The method also needs the supposed definition's
+    // `DefId` to perform comparison.
```

Comment:
`def_def_id` "should" be the `DefId` of supposed definition's parent, i.e. the `scope` (module, enum, struct, trait, or impl) in which we are resolving `use_name` (the name of the the item, variant, field, trait item, or impl item, resp.).

However, I believe this distinction only makes a difference when resolving in modules, so we can probably safely use the semantics in this PR for resolution that happens after `resolve`.